src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -427,6 +427,45 @@ public function siblings()
         return $this->createSubCrawler($this->sibling($this->getNode(0)->parentNode->firstChild));
     }
 
+    public function matches(string $selector): bool
+    {
+        if (!$this->nodes) {
+            throw new \InvalidArgumentException('The current node list is empty.');
+        }
```

Comment:
Is it needed, in other similar cases, we could return `false` here.

Reply:
I'm not sure to understand but I updated the code to return false